### ISSUE TITLE
Deleting Honoree Contact causes unexpected exception

### ISSUE THREAD

**Issue by johnfine** [SalesforceFoundation/Cumulus#1974]
Nulling out the Honoree Contact field on Opportunity causes an TDTM_Opportunity exception and prevents saving a record if error handling is turned off:

1. NPSP Settings-->System Tools-->Error Notifications-->Disable Error Handling checked.
2. Create an Opp with Honoree Contact set to point to a Contact, save the Opp record.
3. Edit the Opp to null out the Honoree Contact field.  Try to save.  Error that prevents the record from being saved:

Error:Apex trigger npsp.TDTM_Opportunity caused an unexpected exception, contact your administrator: npsp.TDTM_Opportunity: execution of BeforeUpdate caused by: System.NullPointerException: Attempt to de-reference a null object: (npsp)

If you uncheck "Disable Error Handling", then using the steps above, the record can be saved correctly.

**Comment by judisohn** [comment]
Interesting @johnfine - out of curiosity, what is the use case that you would want to disable error handling?

**Comment by davidhabib** [comment]
I think the bug is there regardless.  it is just that with error handling turned on, we may be logging it but then continuing with the transaction.  so we want to fix the bug, even if it is only visible when error handling is off.

**Comment by njjc** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x800000003b3JAAQ" target="_blank">W-009518</a>

**Comment by judisohn** [comment]
Also, deleting Honoree Contact lookup should also delete the value in Honoree Name if it matches the name of the Contact lookup (was automatically populated)

**Comment by johnfine** [comment]
@judisohn The use case was a mistake on my part, but here it is, fyi: It related to controlling user perception of errors.  While setting up a gridbuddy grid for data entry by volunteers with limited training, I was trying to avoid an jargon-ish error (as per the recent posts on POU) under certain conditions so the volunteers wouldn't see anything they didn't understand.  I experimented with checking the NPSP Disable Error Handling setting.  Turned out, that had nothing to do with hiding the ugly error msg.  I forgot to enable error handling again, then stumbled across the bug reported here later on.

So I have no real use case for checking Disable Error Handling - I don't even know what exact change in functionality this might cause, or even why the checkbox is available to users at all.
